code-challenges/301/for-each/for-each.test.js

```diff
@@ -0,0 +1,199 @@
+};
+
+const removeElements = (arr, callback) => {
+  for (let i = 0; i <= arr.length; i++) {
+    callback(i, arr);
+  }
+  return arr;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+Write a function named removeWithForEach that produces the same output as challenge 3, but uses forEach.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithForEach = (arr, callback) => {
+  arr.forEach(function(index){
+    callback(index, arr);
+  });
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+Write a function named removeWithAnon that produces the same output as challenges 3 and 4.
+This function should use forEach again, but rather than taking in a callback as an argument, define an anonymous function as the argument to forEach.
+This anonymous function should accept up to three arguments: the element, the index, and the array.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithAnon = (arr) => {
+  arr.forEach(function(index) {
+    if (index % 3 === 2) {
+      arr.pop();
+    }
+  });
+  return arr;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+Write a function named createList that takes in an array of the current store intentory.
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  const groceryList = [];
+  availableItems.forEach(function(item) {
+    if (item.available === true){
+      groceryList.push(item.name);
+    }
+  });
+  return groceryList;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7
+Write a function named fizzbuzz that takes in an array of numbers.
+Iterate over the array using forEach to determine the output based on several rules:
+  - If a number is divisible by 3, add the word "Fizz" to the output array.
+  - If the number is divisible by 5, add the word "Buzz" to the output array.
+  - If the number is divisible by both 3 and 5, add the phrase "Fizz Buzz" to the output array.
+  - Otherwise, add the number to the output array.
+Return the resulting output array.
+------------------------------------------------------------------------------------------------ */
+
+const fizzbuzz = (arr) => {

```

Comment:
You could also structure this like:
`const fizzbuzz = (arr) => {
  const output = [];
  arr.forEach(value => {
    if (value % 5 === 0 && value % 3 === 0) {
      output.push('Fizz Buzz');
    } else if (value % 3 === 0) {
      output.push('Fizz');
    } else if (value % 5 === 0) {
      output.push('Buzz');
    } else {
      output.push(value);
    }
  })

  return output;

}`